decoder.py

```diff
@@ -118,14 +119,30 @@ def forward(self, input, encoder_outs, hidden_init, targets_len):
             # print(f'log_prob shape {log_prob.shape}')
             # print(f'log_prob {log_prob}')
 
+            # FOR UNK POSTPROCESSING
```

Comment:
@annazuo thoughts on moving this out of decoder and instead make decoder's forward() return attention scores so you can compute `best_attn_labels`? my reasoning is that
1. we should separate the functionality of decoder and inference. technically post-processing is part of inference and happens after the model has been trained. (see section 4.3 of the paper)
2. if might be useful for plotting attention mask if we return the attention scores from the decoder too